heron/scheduler-core/src/java/com/twitter/heron/scheduler/UpdateTopologyManager.java

```diff
@@ -152,6 +152,9 @@ private void updateTopology(final PackingPlans.PackingPlan existingProtoPackingP
     // deactivate and sleep
     if (initiallyRunning) {
       deactivateTopology(stateManager, updatedTopology);
+      // Update the topology since the state should have changed from RUNNING to PAUSED
+      updatedTopology =
```

Comment:
Thanks, and sure I'll take a look at it this week. I think that'd require modifying the argument list for deactivateTopology so that the new packing plan is available to construct the final Topology. Is that okay? Another way to do it would be to return a fresh Topology and merge it with the new packing plan in updateTopology().

I first noticed it while reviewing logs for one of my spouts. I noticed that the spout would be activated right after receiving a NewInstanceAssignmentMessage. I then followed it back to the scheduler logs where I saw that the reactivation task would fail because the topology was already running.